test/integration/component/test_deploy_vm_from_snapshot.py

```diff
@@ -0,0 +1,644 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Import Local Modules
+from marvin.codes import FAILED, PASS
+from nose.plugins.attrib import attr
+from marvin.cloudstackTestCase import cloudstackTestCase
+from marvin.cloudstackAPI import seedTemplateFromVmSnapshot
+from marvin.lib.utils import cleanup_resources, validateList
+from marvin.lib.base import (Account,
+                             DiskOffering,
+                             ServiceOffering,
+                             VirtualMachine,
+                             VmSnapshot,
+                             Volume)
+from marvin.lib.common import (get_zone,
+                               get_domain,
+                               get_template)
+import urllib
+import tempfile
+import os
+import time
+from functools import reduce
+
+
+class TestDeployVm(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        testClient = super(TestDeployVm, cls).getClsTestClient()
+        cls.apiclient = testClient.getApiClient()
+        cls._cleanup = []
+        cls.unsupportedHypervisor = False
+        cls.hypervisor = testClient.getHypervisorInfo()
+        if not cls.hypervisor.lower() in "vmware":
+            cls.unsupportedHypervisor = True
+            return
+
+        cls.services = testClient.getParsedTestDataConfig()
+        # Get Zone, Domain and templates
+        cls.domain = get_domain(cls.apiclient)
+        cls.zone = get_zone(cls.apiclient, testClient.getZoneForTests())
+
+        template = get_template(
```

Comment:
@GabrielBrascher  thanks for reviewing it 
1-ExtractVolumeFromVmSnapshot class uses cls.template because test_03_create_volume_from_vmsnapshot uses the self.template, but in class TestDeployVm i am not using template to deploy vm in any test cases thats why i did not declare it as a class variable 
2-Agreed on redundant code in test_02_create_vm_from_vmsnapshots_and_verify_data , ill push modified code as soon as i get some time.